Stores example fix and other improvements (#535)
solidjs/solid-docs-next: src/routes/concepts/stores.mdx

}
```

> When you create a store, it contains the initial state, but _no signals_ are immediately created to track changes to that state.
> These signals are created lazily, meaning they come into existence only when you access the data within a reactive context.

Reviewer: ```suggestion
When a store is created, it starts with the initial state but does _not_ immediately set up signals to track changes.
These signals are created **lazily**, meaning they are only formed when accessed within a reactive context.
```

> When you use this data within a reactive context, such as in a template, computed property, or effect, that is when the signals are created and dependencies are established.
> 
> For example, if you wanted to you wanted to print out every new user, adding the 
> line below won't work because it is not within a tracked scope.

Reviewer: ```suggestion
Once data is used within a reactive context, such as within the return statement of a function, computed property, or an effect, a signal is created and dependencies are established.

For example, if you wanted to print out every new user, adding the console log below will not work because it is not within a tracked scope.
```

Author: @LadyBluenotes I agree that my use of "template" was not specific enough, but I think it needs to be "within the return statement of a **component** function". Just saying function will probably give the wrong impression that any functions return statement is tracked.

```suggestion
Once data is used within a reactive context, such as within the return statement of a component function, computed property, or an effect, a signal is created and dependencies are established.
	
For example, if you wanted to print out every new user, adding the console log below will not work because it is not within a tracked scope.
```

```tsx ins={9}
const App = () => {
	const [store, setStore] = createStore({
		userCount: 3,
		users: [ ... ],
	})

	const addUser = () => { ... }

	console.log(store.users.at(-1)) // This won't work

	return (
		<div>
			<h1>Hello, {store.users[0].username}</h1>
			<p>User count: {store.userCount}</p>
      <button onClick={addUser}>Add user</button>
		</div>
	)
}
```

> Rather, you would have to add this in a tracking scope, such as within a [`createEffect`](/reference/basic-reactivity/create-effect) so that a dependency is established.

Reviewer: ```suggestion
Rather, this would need to be in a tracking scope, like inside a [`createEffect`](/reference/basic-reactivity/create-effect), so that a dependency is established.
```

```tsx del={9} ins={10-12}
const App = () => {
	const [store, setStore] = createStore({
		userCount: 3,
		users: [ ... ],
	})

	const addUser = () => { ... }

	console.log(store.users.at(-1))
	createEffect(() => {
		console.log(store.users.at(-1))
	})

	return (
		<div>
			<h1>Hello, {store.users[0].username}</h1>
			<p>User count: {store.userCount}</p>
      <button onClick={addUser}>Add user</button>
		</div>
	)
}
```

## Modifying store values

Updating values within a store is best accomplished using a setter provided by the `createStore` initialization.
This setter allows for the modification of a specific key and its associated value, following the format `setStore(key, newValue)`:

```jsx "setStore"
const [store, setStore] = createStore({
	userCount: 3,

> 	users: [ ... ],

Reviewer: What is this trying to convey?

Author: @LadyBluenotes it's just a collapsed version of the examples above to reduce space taken up by the repetitive content. It is pretty similar to how editors actually show collapsed lines.

![image](https://github.com/solidjs/solid-docs-next/assets/25469167/af78c1fe-f567-4c65-a2f3-3726d3fbaef3)

])
```

> We could also set update the value of `userCount` whenever a new user is added.

Reviewer: This sentence feels out of place, could you add more context?

Author: @LadyBluenotes I forgot to take out "set" when I reworded this which definitely makes it read weird. It is just to preface the next example. There isn't really a purpose other than to provide a different example of how userCount could be automatically updated based on on the users array. 

I primarily included it because the example was there originally and I figured having another example of a different use case would be good.

```suggestion
The value of `userCount` could also be automatically updated whenever a new user is added to keep it up to date:
```

Let me know what you think about this instead, I don't really know what additional context to add.

```tsx ins={11}
const App = () => {
	const [store, setStore] = createStore({
		userCount: 3,
		users: [ ... ],
	})

	const addUser = () => { ... }

	createEffect(() => {
		console.log(store.users.at(-1))
		setStore("userCount", store.users.length)
	})

	return (
		<div>
			<h1>Hello, {store.users[0].username}</h1>
			<p>User count: {store.userCount}</p>
      <button onClick={addUser}>Add user</button>
		</div>
	)
}
```

<Callout>

>   Separating read and write capabilities to a store presents a valuable debugging advantage.

Reviewer: ```suggestion
  Separating the read and write capabilities of a store provides a valuable debugging advantage.
```

> 	It facilitates the tracking and control of which components can access or change store values.

Reviewer: ```suggestion
	This separation facilitates the tracking and control of the components that are accessing or changing the values.
```
